torch/_inductor/codegen/common.py

```diff
@@ -428,12 +428,10 @@ def __init__(self, name, line):
     def __call__(self):
         # V.kernel may be null since this method may be called for the
         # wrapper codegen where there is no specific kernel.
-        if (
-            self.name
-            not in (
-                V.graph.removed_buffers | getattr(V.kernel, "removed_buffers", set())
-            )
-            and self.name not in V.graph.inplaced_to_remove
+        if self.name not in (
+            V.graph.removed_buffers | getattr(V.kernel, "removed_buffers", set())
+        ) and self.name not in (
+            V.graph.inplaced_to_remove | getattr(V.kernel, "inplaced_to_remove", set())
```

Comment:
It would be cleaner if V.kernel.removed_buffers/inplaced_to_remove were just always defined, so we could read them unconditionally.

This getattr() with defaults pattern can lead to silent correctness issues (rather than loud errors) if someone typos the name.

Reply:
Just recall the reason why I call getattr with default value in the first place:
- I already initialize `removed_buffers` and `inplaced_to_remove` to `set()` in the base kernel class `Kernel`
- However this method may be called outside of a kernel context. i.e., `V.kernel` is a `NullHandler`. The default value for `getattr` is mainly for this.

To resolve the concern regarding getattr with default value, I'll create a NullKernelHandler with `removed_buffers` and `inplaced_to_remove` properly initialized to `set()`. And use NullKernelHandler (rather than NullHandler) if there is no real kernel in the context.